anvio/parsers/hmmer.py

```diff
@@ -0,0 +1,732 @@
+            col_mapping = [int, str, str, str, float, float, str, str, float, str, str, str, str, str, str, str, str, str]
+        else:
+            raise ConfigError("The HMMScan Parser class is not sure if you know what you are doing. You told it that you wanted to "
+                                "parse HMM hits from the program %s, but this class doesn't know how to handle those." % (self.program))
+
+        return col_names, col_mapping
+
+
+    def get_col_info_for_CONTIG_context(self):
+        """Get column names and types for GENE context
+
+        See class docstring for details of the fields for AA sequence search, and DNA sequence search.
+        """
+
+        # 'hmm_target', 'hmm_acc', 'query_id', 'query_acc', 'hmm_from', 'hmm_to', 'alignment_from', 'alignment_to', 'envelope_from', 'envelope_to', 'seq_len', 'strand', 'e_value', 'score', 'bias',]
+        col_names = ['gene_name', 'gene_hmm_id', 'contig_name', 'f', 'hmm_from', 'hmm_to', 'alignment_from', 'alignment_to', 'envelope_from', 'envelope_to', 'f', 'f', 'e_value', 'f', 'f']
+        col_mapping = [str, str, str, str, str, str, int, int, int, int, str, str, float, str, str]
+
+        return col_names, col_mapping
+
+
+    def get_col_info_for_DOMAIN_context(self):
+        """Get column names and types for DOMAIN context
+
+        See class docstring for details of the fields
+        """
+
+        col_info = [
+            ('gene_callers_id', str), # target name
+            ('f',               str), # accession
+            ('gene_length',     str), # tlen
+            ('hmm_name',        str), # query name
+            ('hmm_id',          str), # accession
+            ('hmm_length',      str), # qlen
+            ('evalue',          str), # E-value (full sequence)
+            ('bitscore',        str), # score (full sequence)
+            ('bias',            str), # bias (full sequence)
+            ('match_num',       str), # # (this domain)
+            ('num_matches',     str), # of (this domain)
+            ('dom_c_evalue',    str), # c-Evalue (this domain)
+            ('dom_i_evalue',    str), # i-Evalue (this domain)
+            ('dom_bitscore',    str), # score (this domain)
+            ('dom_bias',        str), # bias (this domain)
+            ('hmm_start',       str), # from (hmm coord)
+            ('hmm_stop',        str), # to (hmm coord)
+            ('gene_start',      str), # from (ali coord)
+            ('gene_stop',       str), # to (ali coord)
+            ('f',               str), # from (env coord)
+            ('f',               str), # to (env coord)
+            ('mean_post_prob',  str), # acc
+        ]
+
+        return list(zip(*col_info))
+
+
+    def get_search_results(self, noise_cutoff_dict = None):
+        """Goes through the hits provided by `hmmscan` and generates an annotation dictionary with the relevant information about each hit.
```

Comment:
Evan, I just wanted to let you know that I made some changes to this function in master last night (so that it could return a dictionary of bitscores when requested; this is related to our desire to compute new KEGG score thresholds). It was basically a crappy patch to get the functionality that I wanted out of the current hmmscan parser design.

That functionality will need to be incorporated somehow into your HMMER overhaul (which is looking very nice, by the way, and was much-needed :). Probably this is a good opportunity to turn it from a crappy patch into a robust design enabling bitscore retrieval. I also agree with you that the pre-processing of the hmmer output needs to be moved out of the driver class, so perhaps we could do that at the same time. 

Do you want to have a meeting about it at some point?

Reply:
@ivagljiva thanks for your message. I will try and merge master into this branch and resolve merge conflicts. Once I do that, could you please just double check that I did not break the KEGG stuff?

I'm glad you agree that we should not preprocess the HMMER output. I think it would be worth discussing how to do that, however I think we should do it in a different branch and perhaps different time. Having spent a lot of time getting that code to do what I want, I would prefer stepping away from it for the time being because I found it very frustrating to work with.

Reply:
Sure! I'll take a look.

And that is completely understandable. You've spent heaps of effort on this already :)